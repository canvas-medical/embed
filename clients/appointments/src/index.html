<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>Appointments</title>
    <meta name="viewport" content="width=device-width,initial-scale=1" />
    <style>
      @import url('https://fonts.googleapis.com/css2?family=Roboto:wght@400;700&display=swap');

      body {
        margin: 0;
      }
    </style>
  </head>
  <body>
    <div id="embed-root"></div>
    <script src="./appointments.js" type="text/javascript"></script>
    <script type="text/javascript">
      const callbacks = {
        onClick: (e, config) => {
          if (e && e.target) {
            // Prescribed method for getting the element clicked
            // const closestTarget = e.target.closest('[data-analytics-id]')
            // console.log(
            //   closestTarget.dataset,
            //   'e.target.dataset',
            //   config,
            //   'config'
            // )
          }
        },
        onChange: (e, config) => {
          if (e && e.target) {
            // const closestTarget = e.target.closest('[data-analytics-id]')
            // console.log(
            //   closestTarget.dataset,
            //   'e.target.dataset',
            //   config,
            //   'config'
            // )
          }
        },
        onError: (error, msg) => {
          // console.error('error', error, msg)
        },
        onLoad: () => {},
      }

      const config = {
        apiKey: 'API_KEY',
        patientId: 'PATIENT_ID',
        providerIds: ['PROVIDER_ID'],
      }

      const apiBaseUrl = 'http://localhost:3000'

      fetch(
        `${apiBaseUrl}/Auth?` +
          new URLSearchParams({
            key: config.apiKey,
            patient: config.patientId,
          })
      )
        .then(res => res.json())
        .then(data => {
          const patientKey = data.patient_key

          Appointments.init({
            api: apiBaseUrl,
            bailoutURL: 'https://canvasmedical.com',
            locationId: '1',
            patientId: config.patientId,
            patientKey: patientKey,
            rootId: 'embed-root',
            callbacks,
            // brandColor: '',
            // accentColor: '',
            // fontFamily: 'Arial, serif',
          })
        })
    </script>
  </body>
</html>
