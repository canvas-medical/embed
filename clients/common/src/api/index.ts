export * from './get-time-slots'
