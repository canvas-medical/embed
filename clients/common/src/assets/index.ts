export * from './arrow-back'
export * from './warning'
