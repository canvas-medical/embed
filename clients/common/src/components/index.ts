export * from './backdrop'
export * from './boxes'
export * from './buttons'
export * from './error'
export * from './header'
export * from './iconography'
export * from './loader'
export * from './typograhpy'
