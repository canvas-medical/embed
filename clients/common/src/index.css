*,
*:before,
*:after {
  box-sizing: border-box;
}

h1,
h2,
h3,
p,
ol,
ul,
fieldset,
legend {
  margin: 0;
  padding: 0;
  font-weight: normal;
}

button {
  cursor: pointer;
}

ul {
  list-style: none;
}
