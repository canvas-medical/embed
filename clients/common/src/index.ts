import css from './index.css'

export * from './assets'
export * from './components'
export * from './utils'
export { css }
