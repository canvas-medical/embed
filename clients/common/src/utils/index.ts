export * from './dates'
export * from './styles'
export * from './types'
