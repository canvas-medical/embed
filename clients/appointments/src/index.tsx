import { h, render } from 'preact'
import { StyleSheetManager } from 'styled-components'
import { css, Error, generateColors } from '@canvas-medical/embed-common'
import { App } from './App'
import { IInitializerProps, IAppointmentProps, hasAllValues } from './utils'

export const Appointments = (props: IAppointmentProps) => {
  const {
    api,
    bailoutURL,
    locationId,
    patientId,
    patientKey,
    providerIds,
    brandColor,
    accentColor,
    shadowRoot,
    customFont,
  } = props
  const colors = generateColors(brandColor, accentColor)
  const allValuesProvided = hasAllValues(props)

  return (
    // Ignoring type mismatch error on target - ShadowRoot is an acceptable type
    // @ts-ignore
    <StyleSheetManager target={shadowRoot}>
      {!allValuesProvided.length ? (
        <App
          api={api}
          bailoutURL={bailoutURL}
          colors={colors}
          locationId={locationId}
          patientId={patientId}
          patientKey={patientKey}
          providerIds={providerIds}
          shadowRoot={shadowRoot}
          customFont={customFont}
        />
      ) : (
        <Error errorMessages={allValuesProvided} />
      )}
    </StyleSheetManager>
  )
}

export const init = ({
  api,
  bailoutURL,
  locationId,
  patientId,
  patientKey,
  providerIds,
  brandColor,
  accentColor,
  rootId,
  customFont,
}: IInitializerProps) => {
  const appRoot = document.querySelector(`#${rootId}`)

  if (!appRoot) {
    console.error('App root could not be found. Check your rootId')
    return null
  }

  appRoot.attachShadow({
    mode: 'open',
  })

  if (!appRoot.shadowRoot) {
    console.error('Shadow root could not be attached')
    return null
  }

  const styleTag = document.createElement('style')
  styleTag.innerHTML = css
  appRoot.shadowRoot.appendChild(styleTag)

  render(
    <Appointments
      api={api}
      bailoutURL={bailoutURL}
      locationId={locationId}
      patientId={patientId}
      patientKey={patientKey}
      providerIds={providerIds}
      brandColor={brandColor}
      accentColor={accentColor}
      shadowRoot={appRoot.shadowRoot}
      customFont={customFont}
    />,
    appRoot.shadowRoot
  )
}
