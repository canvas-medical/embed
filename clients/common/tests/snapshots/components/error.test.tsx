import { h } from 'preact'
import { render } from 'preact-render-to-string'
import { Error } from '../../../src'

describe('Error', () => {
  it('renders with a single error message', () => {
    const error = 'This is a single error'

    expect(render(<Error errorMessages={error} />)).toMatchSnapshot()
  })

  it('renders with multiple error messages', () => {
    const error = [
      'This is the first error error',
      'This is the second error',
      'This is the third error',
    ]

    expect(render(<Error errorMessages={error} />)).toMatchSnapshot()
  })
})
