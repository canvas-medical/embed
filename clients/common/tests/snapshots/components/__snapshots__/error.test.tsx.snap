// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders with a single error message 1`] = `"<div class=\\"sc-hKwDye bgTSDk\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" enable-background=\\"new 0 0 24 24\\" height=\\"48px\\" viewBox=\\"0 0 24 24\\" width=\\"48px\\" fill=\\"#D02121\\"><g><rect fill=\\"none\\" height=\\"24\\" width=\\"24\\"></rect></g><g><g><g><path d=\\"M12,5.99L19.53,19H4.47L12,5.99 M12,2L1,21h22L12,2L12,2z\\"></path><polygon points=\\"13,16 11,16 11,18 13,18\\"></polygon><polygon points=\\"13,10 11,10 11,15 13,15\\"></polygon></g></g></g></svg><div class=\\"sc-hKwDye fojPTN\\"><span font-size=\\"0.875rem\\" class=\\"sc-iqseJM fDGtm\\">This is a single error</span></div></div>"`;

exports[`Error renders with multiple error messages 1`] = `"<div class=\\"sc-hKwDye bgTSDk\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" enable-background=\\"new 0 0 24 24\\" height=\\"48px\\" viewBox=\\"0 0 24 24\\" width=\\"48px\\" fill=\\"#D02121\\"><g><rect fill=\\"none\\" height=\\"24\\" width=\\"24\\"></rect></g><g><g><g><path d=\\"M12,5.99L19.53,19H4.47L12,5.99 M12,2L1,21h22L12,2L12,2z\\"></path><polygon points=\\"13,16 11,16 11,18 13,18\\"></polygon><polygon points=\\"13,10 11,10 11,15 13,15\\"></polygon></g></g></g></svg><div class=\\"sc-hKwDye fojPTN\\"><span font-size=\\"0.875rem\\" class=\\"sc-iqseJM fDGtm\\">This is the first error error</span><span font-size=\\"0.875rem\\" class=\\"sc-iqseJM fDGtm\\">This is the second error</span><span font-size=\\"0.875rem\\" class=\\"sc-iqseJM fDGtm\\">This is the third error</span></div></div>"`;
